### PR TITLE
Add FAQ section in README to explain usage of Generics

### PR DESCRIPTION
**Pull request by browniebroke**
# I have made things!

I recently came across the fact that this library adds support for Generics in DRF serializers, which makes type checking a lot better. It might be obvious to the people who are familiar with type checking, but it wasn't obvious to me that this library supported that.

## Related issues

Refs #164

## Follow up

I think it's also supported for GenericViewSet, but requires [a bit more extra setup](https://github.com/typeddjango/djangorestframework-stubs/issues/255). If that's correct, I can also add a paragraph/section about that, although I'd like to get feedback on this change first.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,6 +26,20 @@ plugins =
...
 in your `mypy.ini` file.
 
+## FAQ
+
+### Model instance is inferred as `Any` instead of my `Model` class
+
+This library adds support for [Generics](https://peps.python.org/pep-0484/#generics) in DRF. For example, you can now write:
+
+```python
+class MyModelSerializer(serializers.ModelSerializer[MyModel]):
```
Comment (sobolevn) [inline README.md:36]: Do we need to monkeypatch it?

Comment (browniebroke) [reply README.md:36]: Do you mean call `django_stubs_ext.monkeypatch()`? I'm pretty sure it's not needed, worked without it in my project.

Comment (browniebroke) [reply README.md:36]: The `__class_getitem__` method was added https://github.com/encode/django-rest-framework/pull/7385 which was released in DRF 3.12

**Comment by browniebroke** [comment]
> I think it's also supported for GenericViewSet, but requires [a bit more extra setup](https://github.com/typeddjango/djangorestframework-stubs/issues/255).

I tried to use it in my project but it didn't work. After a quick search, the monkeypatching might not be working [so well for DRF](https://github.com/typeddjango/djangorestframework-stubs/issues/299#issuecomment-1343952084), depending on your DRF settings.

From what I understand, it's not properly fixable in any of the projects from the @typeddjango org, and it should be better once DRF 3.15 is out, which will include [this change](https://github.com/encode/django-rest-framework/pull/8825). I suggest to not advertise this feature for views until 3.15 is out, and it will need a note about the minimum version.

**Comment by sobolevn** [comment]
Do we have a ETA for the release?

**Comment by browniebroke** [comment]
There is a [PR open](https://github.com/encode/django-rest-framework/pull/8794) to prepare the release notes. It's been open for a while now and it looks like it's all ready. Probably just waiting on a maintainer to find enough free time to do it.

File: README.md [hunk 2]
```diff
@@ -26,6 +26,20 @@ plugins =
 
 in your `mypy.ini` file.
 
+## FAQ
+
+### Model instance is inferred as `Any` instead of my `Model` class
+
+This library adds support for [Generics](https://peps.python.org/pep-0484/#generics) in DRF. For example, you can now write:
```
Comment (intgr) [inline README.md:33]: I would rephrase this:

```suggestion
When subclassing `ModelSerializer`, add a [type argument](https://peps.python.org/pep-0484/#generics) to type-hint the related model class, for example:
```

Feel free to tweak this further.
